MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -1697,36 +1697,62 @@ static void signalAndUntrack(const MVKSwapchainSignaler& signaler) {
 		default:
 			break;
 	}
+	
+	// If only stencil aspect is requested, possibly change to stencil-only format.
+	if (aspectMask == VK_IMAGE_ASPECT_STENCIL_BIT) {
+		if (_mtlPixFmt == MTLPixelFormatDepth32Float_Stencil8) {
+			_mtlPixFmt = MTLPixelFormatX32_Stencil8;
+		}
+#if MVK_MACOS
+		else if (_mtlPixFmt == MTLPixelFormatDepth24Unorm_Stencil8) {
+			_mtlPixFmt = MTLPixelFormatX24_Stencil8;
+		}
+#endif
+	}
+	
+	// Due to differences in Metal and Vulkan specification for sampling depth/stencil textures into vec4, we need to
+	// provide the correct mapping from Vulkan to Metal
+	// Metal states:
+	// "For a texture with depth or stencil pixel format (such as MTLPixelFormatDepth24Unorm_Stencil8 or MTLPixelFormatStencil8),
+	// the default value for an unspecified component is undefined." which means all values but R will be undefined.
+	// Vulkan states: https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#textures-conversion-to-rgba
+	// So we need to ensure that G and B are set to 0 and A to 1 for the swizzle in Metal
+	if (mvkIsAnyFlagEnabled(aspectMask, VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT) &&
+		mvkIsAnyFlagEnabled(_imageView->_usage, (VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT)) &&
+		enableSwizzling()) {
+		if (_componentSwizzle.r == VK_COMPONENT_SWIZZLE_A) {
+			_componentSwizzle.r = VK_COMPONENT_SWIZZLE_ONE;
+		}
+		else if (_componentSwizzle.r == VK_COMPONENT_SWIZZLE_G || _componentSwizzle.r == VK_COMPONENT_SWIZZLE_B) {
+			_componentSwizzle.r = VK_COMPONENT_SWIZZLE_ZERO;
+		}
+		if (_componentSwizzle.g == VK_COMPONENT_SWIZZLE_A) {
+			_componentSwizzle.g = VK_COMPONENT_SWIZZLE_ONE;
+		}
+		else if (_componentSwizzle.g == VK_COMPONENT_SWIZZLE_G || _componentSwizzle.g == VK_COMPONENT_SWIZZLE_B || _componentSwizzle.g == VK_COMPONENT_SWIZZLE_IDENTITY) {
+			_componentSwizzle.g = VK_COMPONENT_SWIZZLE_ZERO;
+		}
+		if (_componentSwizzle.b == VK_COMPONENT_SWIZZLE_A) {
+			_componentSwizzle.b = VK_COMPONENT_SWIZZLE_ONE;
+		}
+		else if (_componentSwizzle.b == VK_COMPONENT_SWIZZLE_G || _componentSwizzle.b == VK_COMPONENT_SWIZZLE_B || _componentSwizzle.b == VK_COMPONENT_SWIZZLE_IDENTITY) {
+			_componentSwizzle.b = VK_COMPONENT_SWIZZLE_ZERO;
+		}
+		if (_componentSwizzle.a == VK_COMPONENT_SWIZZLE_A || _componentSwizzle.a == VK_COMPONENT_SWIZZLE_IDENTITY) {
+			_componentSwizzle.a = VK_COMPONENT_SWIZZLE_ONE;
+		}
+		else if (_componentSwizzle.a == VK_COMPONENT_SWIZZLE_G || _componentSwizzle.a == VK_COMPONENT_SWIZZLE_B ) {
```

Comment:
Minor style issue for consistency. On each of these. please pull the `else` up to the bracket line:  `} else {`. 

Line 1707 is fine because of the conditional compile.

Reply:
Fixed